pwiz_tools/Skyline/Controls/Databinding/DataboundGridControl.cs

```diff
@@ -663,149 +663,22 @@ private void heatMapContextMenuItem_Click(object sender, EventArgs e)
-                    columnGroups.Add(new ClusterGraphResults.ColumnGroup(
-                        clusteredResults.ColumnGroupDendrogramDatas[iGroup].DendrogramData, groupHeaders));
-                    for (int iProperty = 0; iProperty < series.PropertyIndexes.Count; iProperty++)
-                    {
-                        var columnHeaders = groupColumnHeaders.Prepend(series)
-                            .Select(s => s.PropertyDescriptors[iProperty]).ToList();
-                        var cellLocator = CellLocator.ForColumn(columnHeaders, clusteredResults.ItemProperties);
-                        cellLocators.Add(cellLocator);
-                    }
-                }
-            }
-            for (int iRow = 0; iRow < clusteredResults.RowCount; iRow++)
-            {
-                var rowItem = clusteredResults.RowItems[iRow];
-                var rowColors = new List<Color>();
-                var rowHeaderParts = new List<object>();
-                foreach (var rowHeader in clusteredResults.ClusteredProperties.RowHeaders)
-                {
-                    rowHeaderParts.Add(rowHeader.GetValue(rowItem));
-                    rowColors.Add(colorScheme.GetColor(rowHeader, rowItem) ?? Color.Transparent);
-                }
-
-                string rowCaption = CaptionComponentList.SpaceSeparate(rowHeaderParts)
-                    .GetCaption(DataSchemaLocalizer.INVARIANT);
-                rowHeaders.Add(new ClusterGraphResults.Header(rowCaption, rowColors));
-                int iCol = 0;
-                foreach (var series in columnValues)
-                {
-                    foreach (var color in colorScheme.GetSeriesColors(series, rowItem))
-                    {
-                        var point = new ClusterGraphResults.Point(iRow, iCol, color);
-                        var skylineDocNode = cellLocators[iCol].GetSkylineDocNode(rowItem);
-                        if (skylineDocNode != null)
-                        {
-                            point = point.ChangeIdentityPath(skylineDocNode.IdentityPath);
-                        }
+            return new ClusterInput(BindingListSource.ViewInfo.DataSchema.DataSchemaLocalizer, BindingListSource.ReportResults, BindingListSource.ClusteringSpec);
+        }
 
-                        var replicate = cellLocators[iCol].GetReplicate(rowItem);
-                        if (replicate != null)
-                        {
-                            point = point.ChangeReplicateName(replicate.Name);
-                        }
-                        points.Add(point);
-                        iCol++;
-                    }
-                }
-            }
-            var graphResults = new ClusterGraphResults(clusteredResults.RowDendrogramData?.DendrogramData, rowHeaders, columnGroups, points);
-            var heatMapGraph = new HierarchicalClusterGraph()
+        public bool ShowHeatMap()
+        {
+            var heatMapGraph = new HierarchicalClusterGraph

```

Comment:
They are dockable now, yes, but it looks like we still just create a new one whenever one is shown, and I don't see any code to preserve them in the layout, if I do File > Save. Do they get properly preserved in the .sky.view file and restored? Do other group comparison windows?